### PR TITLE
nightmare: add plugin

### PR DESCRIPTION
**Pull request by ASwatek**
shows what to pray against, highlights shadows, more things planned for later today

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -23,3 +23,5 @@ runelite-client/dependencies.txt
 .staging/
 .vscode
 .factorypath
+/runelite-client/src/main/resources/open.osrs.properties
```
Comment (ThatGamerBlue) [inline .gitignore]: no

File: runelite-client/src/main/java/net/runelite/client/plugins/nightmare/NightmarePlugin.java [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+import net.runelite.client.plugins.nightmare.NightmareAttack;
+import net.runelite.client.plugins.nightmare.NightmareOverlay;
+import net.runelite.client.plugins.nightmare.NightmarePrayerOverlay;
+import net.runelite.client.ui.overlay.OverlayManager;
+import net.runelite.client.ui.overlay.components.PanelComponent;
+
+
+@PluginDescriptor(
+        name = "Nightmare of Ashihama",
+        description = "make braindead shit boss more braindead",
```
Comment (ThatGamerBlue) [inline runelite-client/src/main/java/net/runelite/client/plugins/nightmare/NightmarePlugin.java]: no

File: runelite-client/src/main/resources/open.osrs.properties [hunk 3]
```diff
@@ -1,15 +1,15 @@
 runelite.version=@project.version@
 runescape.version=@rs.version@
-open.osrs.discord.appid=627741263881568257
```
Comment (ThatGamerBlue) [inline runelite-client/src/main/resources/open.osrs.properties]: what even happened in this file

Comment (ASwatek) [reply runelite-client/src/main/resources/open.osrs.properties]: > 
> 
> what even happened in this file

that wasnt supposed to be submitted, i thought i had it in my gitignore but i guess not, im absolutely garbage with version control

File: runelite-client/src/main/java/net/runelite/client/plugins/nightmare/NightmarePlugin.java (hunk 2 above)
Comment (ASwatek) [reply runelite-client/src/main/java/net/runelite/client/plugins/nightmare/NightmarePlugin.java]: > 
> 
> no

what is this no referring to

**Comment by emiljensen2** [comment]
The prayer thing works for like 1 phase inside then it stops working until you leave and rejoin the nightmare.

**Comment by ASwatek** [comment]
> 
> 
> The prayer thing works for like 1 phase inside then it stops working until you leave and rejoin the nightmare.

that should be fixed now.  up until i saw one of you mention it in the server, i was just testing the first 30s of the boss.  after i made some changes, i stayed for whole boss and pray stayed whole time, so it should be good now

File: runelite-client/src/main/java/net/runelite/client/plugins/nightmare/NightmarePlugin.java (hunk 2 above)
Comment (ThatGamerBlue) [reply runelite-client/src/main/java/net/runelite/client/plugins/nightmare/NightmarePlugin.java]: the description

File: runelite-client/src/main/resources/open.osrs.properties (hunk 3 above)
Comment (uupiits) [reply runelite-client/src/main/resources/open.osrs.properties]: Weird, still turn off after 1st phase for me

**Comment by uupiits** [comment]
> > The prayer thing works for like 1 phase inside then it stops working until you leave and rejoin the nightmare.
> 
> that should be fixed now. up until i saw one of you mention it in the server, i was just testing the first 30s of the boss. after i made some changes, i stayed for whole boss and pray stayed whole time, so it should be good now

Weird, still turns off after 1st phase for me

**Comment by ASwatek** [comment]
> 
> 
> > > The prayer thing works for like 1 phase inside then it stops working until you leave and rejoin the nightmare.
> > 
> > 
> > that should be fixed now. up until i saw one of you mention it in the server, i was just testing the first 30s of the boss. after i made some changes, i stayed for whole boss and pray stayed whole time, so it should be good now
> 
> Weird, still turns off after 1st phase for me

i tested two kills with long killtimes and the prayers never glitched up after this commit

**Comment by xKylee** [comment]
@ASwatek can you join our discord please?

File: runelite-client/src/main/java/net/runelite/client/plugins/nightmare/NightmareOverlay.java [hunk 4]
```diff
@@ -0,0 +1,139 @@
+package net.runelite.client.plugins.nightmare;
+
+import net.runelite.api.Client;
+import net.runelite.api.GraphicsObject;
+import net.runelite.api.NPC;
+import net.runelite.api.Perspective;
+import net.runelite.api.coords.LocalPoint;
+import net.runelite.client.plugins.zulrah.ImagePanelComponent;
+import net.runelite.client.ui.overlay.*;
```
Comment (xKylee) [inline runelite-client/src/main/java/net/runelite/client/plugins/nightmare/NightmareOverlay.java]: fix wildcard import

File: runelite-client/src/main/java/net/runelite/client/plugins/nightmare/NightmareOverlay.java [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+import net.runelite.client.ui.overlay.components.PanelComponent;
+import net.runelite.client.ui.overlay.components.TextComponent;
+
```
Comment (xKylee) [inline runelite-client/src/main/java/net/runelite/client/plugins/nightmare/NightmareOverlay.java]: remove whitespace

File: runelite-client/src/main/java/net/runelite/client/plugins/nightmare/NightmareOverlay.java [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.awt.*;
```
Comment (xKylee) [inline runelite-client/src/main/java/net/runelite/client/plugins/nightmare/NightmareOverlay.java]: fix wildcard import

File: runelite-client/src/main/java/net/runelite/client/plugins/nightmare/NightmareOverlay.java [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+//					if(npc.getId() == 9417) bgCol = Color.green;
+//					else if(npc.getId() == 9419) bgCol = Color.red;
+//					else if(npc.getId() == 9421) bgCol = Color.yellow;
+//					else if(npc.getId() == 9423) bgCol = Color.green;
+//					imagePanelComponent.render(graphics);
+//				}
+//			}
+//		}
+
+		if (!client.isInInstancedRegion()) {
```
Comment (xKylee) [inline runelite-client/src/main/java/net/runelite/client/plugins/nightmare/NightmareOverlay.java]: you would be better off checking to see if you're not in the nightmare region.

File: runelite-client/src/main/java/net/runelite/client/plugins/nightmare/NightmareAttack.java [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+	}
+
+	public int getAnimation() {
+		return animation;
+	}
+
+	public Prayer getPrayer() {
+		return prayer;
+	}
+}
```
Comment (xKylee) [inline runelite-client/src/main/java/net/runelite/client/plugins/nightmare/NightmareAttack.java]: ill be honest, i don't like the way this is done. you would be better off creating an attack style enum and incorporating it in a similar way to how fightcaves do.

File: runelite-client/src/main/java/net/runelite/client/plugins/nightmare/NightmarePlugin.java [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+import net.runelite.api.Client;
+import net.runelite.api.NPC;
+import net.runelite.api.events.GameTick;
+import net.runelite.client.eventbus.Subscribe;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+import net.runelite.client.plugins.PluginType;
+import net.runelite.client.ui.overlay.OverlayManager;
+import net.runelite.client.ui.overlay.components.PanelComponent;
+
```
Comment (xKylee) [inline runelite-client/src/main/java/net/runelite/client/plugins/nightmare/NightmarePlugin.java]: remove whitespace.

File: runelite-client/src/main/java/net/runelite/client/plugins/nightmare/NightmarePlugin.java [hunk 10]
```diff
@@ -0,0 +1,125 @@
...
+import javax.annotation.Nullable;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.awt.*;
```
Comment (xKylee) [inline runelite-client/src/main/java/net/runelite/client/plugins/nightmare/NightmarePlugin.java]: remove wildcard

File: runelite-client/src/main/java/net/runelite/client/plugins/nightmare/NightmarePlugin.java [hunk 11]
```diff
@@ -0,0 +1,125 @@
...
+import java.awt.image.BufferedImage;
+import java.util.ArrayList;
+
+//import static net.runelite.api.NpcID.nightmarewhenitsupdated;                     //TODO: this here is the thing to change
+
```
Comment (xKylee) [inline runelite-client/src/main/java/net/runelite/client/plugins/nightmare/NightmarePlugin.java]: remove whitespace.

File: runelite-client/src/main/java/net/runelite/client/plugins/nightmare/NightmarePrayerOverlay.java [hunk 12]
```diff
@@ -0,0 +1,93 @@
...
+import net.runelite.client.ui.overlay.OverlayLayer;
+import net.runelite.client.ui.overlay.OverlayPosition;
+import net.runelite.client.ui.overlay.OverlayPriority;
+import net.runelite.client.ui.overlay.components.ComponentConstants;
+import net.runelite.client.ui.overlay.components.ImageComponent;
+import net.runelite.client.ui.overlay.components.PanelComponent;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.awt.*;
```
Comment (xKylee) [inline runelite-client/src/main/java/net/runelite/client/plugins/nightmare/NightmarePrayerOverlay.java]: remove wildcard.

File: runelite-client/src/main/java/net/runelite/client/plugins/nightmare/NightmarePrayerOverlay.java [hunk 13]
```diff
@@ -0,0 +1,93 @@
...
+import net.runelite.api.NPC;
+import net.runelite.client.game.SpriteManager;
+import net.runelite.client.ui.overlay.Overlay;
+import net.runelite.client.ui.overlay.OverlayLayer;
+import net.runelite.client.ui.overlay.OverlayPosition;
+import net.runelite.client.ui.overlay.OverlayPriority;
+import net.runelite.client.ui.overlay.components.ComponentConstants;
+import net.runelite.client.ui.overlay.components.ImageComponent;
+import net.runelite.client.ui.overlay.components.PanelComponent;
+
```
Comment (xKylee) [inline runelite-client/src/main/java/net/runelite/client/plugins/nightmare/NightmarePrayerOverlay.java]: remove whitespace.

**Comment by genetic-soybean** [comment]
Code seems to work for multiple kills. However the protect prayer indicator is wrong if you are cursed.

**Comment by ASwatek** [comment]
> @ASwatek can you join our discord please?

I just joined, and will make these changes shortly.



> Code seems to work for multiple kills. However the protect prayer indicator is wrong if you are cursed.

I know about that, and I also know the curse lasts for 5 of her attacks, and I'm working on figuring out how to change the icon for those 5 hits.  My current idea will cause problems if she phases during the curse (which cancels it), but I think I have a workaround for that as well.  I'll let you know when I change that.

File: runelite-client/src/main/java/net/runelite/client/plugins/nightmare/NightmareOverlay.java (hunk 7 above)
Comment (ASwatek) [reply runelite-client/src/main/java/net/runelite/client/plugins/nightmare/NightmareOverlay.java]: When examining the regions, I didn't see any green number highlighted, so I didn't know which region number to use.  I'm quite new to this, so maybe I'm missing something.

**Comment by ReeeMan** [comment]
@ASwatek use OnChatMessage

When the curse starts it's 'The Nightmare has cursed you, shuffling your prayers!'

And when it ends `You feel the effects of the nightmare's curse wear off`

**Comment by xKylee** [comment]
due to the rework, and our stance of going compliant, this will not be something we add.

**Pull request closed without merging**